src/XML/src/FMI3/fmi3_xml_model_description.c

```diff
@@ -405,197 +424,245 @@ int fmi3_xml_is_valid_model_ID(const char* str) {
+                fmi3_xml_set_attr_enum(context, fmi3_xml_elmID_fmiModelDescription, fmi_attr_id_variableNamingConvention, 0, (unsigned*)&(md->namingConvension),
+                                       fmi3_naming_enu_flat, namingConventionMap);
+        return (ret);
     }
     else {
         /* check that fmuKind is defined and that model identifiers are valid */
-        if(md->fmuKind == fmi3_fmu_kind_unknown) {
+        if (md->fmuKind == fmi3_fmu_kind_unknown) {
             fmi3_xml_parse_fatal(
                 context,
                 "Neither ModelExchange, CoSimulation nor ScheduledExecution element were parsed correctly. FMU kind not known.");
             return -1;
         }
-        if( (md->fmuKind & fmi3_fmu_kind_me) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_ME(md))) {
-            fmi3_xml_parse_error(context, "Model identifier '%s' is not valid (must be a valid C-identifier)", fmi3_xml_get_model_identifier_ME(md));
+        if ((md->fmuKind & fmi3_fmu_kind_me) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_ME(md))) {
+            fmi3_xml_parse_error(context,
+                    "ModelExchange: Model identifier '%s' is not valid (must be a valid C-identifier)",
+                    fmi3_xml_get_model_identifier_ME(md));
             return -1;
         }
-        if( (md->fmuKind & fmi3_fmu_kind_cs) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_CS(md))) {
-            fmi3_xml_parse_error(context, "Model identifier '%s' is not valid (must be a valid C-identifier)", fmi3_xml_get_model_identifier_CS(md));
+        if ((md->fmuKind & fmi3_fmu_kind_cs) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_CS(md))) {
+            fmi3_xml_parse_error(context,
+                    "CoSimulation: Model identifier '%s' is not valid (must be a valid C-identifier)",
+                    fmi3_xml_get_model_identifier_CS(md));
             return -1;
         }
-        if( (md->fmuKind & fmi3_fmu_kind_se) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_SE(md))) {
-            fmi3_xml_parse_error(context, "Model identifier '%s' is not valid (must be a valid C-identifier)", fmi3_xml_get_model_identifier_SE(md));
+        if ((md->fmuKind & fmi3_fmu_kind_se) && !fmi3_xml_is_valid_model_ID(fmi3_xml_get_model_identifier_SE(md))) {
+            fmi3_xml_parse_error(context,
+                    "ScheduledExecution: Model identifier '%s' is not valid (must be a valid C-identifier)",
+                    fmi3_xml_get_model_identifier_SE(md));
             return -1;
         }
-        if(!md->modelStructure) {
+        if (!md->modelStructure) {
             fmi3_xml_parse_fatal(context, "No model structure information available. Cannot continue.");
             return -1;
         }
         return 0;
     }
 }
 
+#define FMI3_XML_REPLACE_WITH_CH(S) '_',
```

Comment:
What does this do, I get a sense this invokes some macro combo and is possibly related to that single "\" linear early & that could use some more explanation as to what it does.